embedchain/embedder/clarifai.py

```diff
@@ -0,0 +1,52 @@
+import os
+from typing import Optional, Union
+
+from embedchain.config import BaseEmbedderConfig
+from embedchain.embedder.base import BaseEmbedder
+
+from chromadb import EmbeddingFunction, Embeddings
+
+class ClarifaiEmbeddingFunction(EmbeddingFunction):
+    def __init__(self, config: BaseEmbedderConfig) -> None:
+        super().__init__()
+        try:
+            from clarifai.client.model import Model
+            from clarifai.client.input import Inputs
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError(
+                "The required dependencies for ClarifaiEmbeddingFunction are not installed."
+                'Please install with `pip install --upgrade "embedchain[clarifai]"`'
+            ) from None
+        self.config = config
+        self.api_key = config.api_key or os.getenv("CLARIFAI_PAT")
+        self.model = config.model or "https://clarifai.com/clarifai/main/models/BAAI-bge-base-en-v15"
```

Comment:
Is there a way to verify only embedding model URL is given here?

Reply:
> Is there a way to verify only embedding model URL is given here?

I guess the only way it is catched is by the try/except block. since we are retrieving only data.embeddings, we'll get an error at that point. By default we don't have any validating function to check if the model is embedder or not right ?!

Reply:
We can by loading model info and checking `model_info.model_type_id` . But yeah, it will cause latency. Fine 👍